### ISSUE TITLE
Node v10.13.0 fs.mkdir recursive option not supported

### ISSUE THREAD

**Issue by anthify** [tschaub/mock-fs#257]
Hey,

It appears there is no support for the seemingly new `fs.mkdir` option `recursive` which recursively builds paths.
```javascript
// Creates /tmp/a/apple, regardless of whether `/tmp` and /tmp/a exist.
fs.mkdir('/tmp/a/apple', { recursive: true }, (err) => {
  if (err) throw err;
}); 
```
Going by this issues https://github.com/tschaub/mock-fs/issues/254#issuecomment-425573723 I guess this might not be implemented as it's a node v10+ feature?

Thanks 😃

**Comment by csomakk** [comment]
have the sane ussue... :/

**Comment by csomakk** [comment]
so instead using
fs.mkdirSync(foldersToCreate[i], {recursive: true});
you can use
                    var parentFolder = destinationDir.replace(/\/[^\/]+\/?$/, '');
                    var foldersToCreate = [destinationDir];
                    while (!fs.existsSync(parentFolder)) {
                        foldersToCreate.push(parentFolder);
                        parentFolder = parentFolder.replace(/\/[^\/]+\/?$/, '');
                    }
                    foldersToCreate = foldersToCreate.reverse();
                    for (var i = 0; i < foldersToCreate.length; i++) {
                        fs.mkdirSync(foldersToCreate[i], {recursive: true});
                    }

**Comment by hsiwkin** [comment]
Hi, I've noticed that the new version with this commit wasn't published on npm. 
To get this fix one has to install it from github (this line in package.json will do the trick: `"mock-fs": "git+https://github.com/tschaub/mock-fs.git"`)

**Comment by tschaub** [comment]
Fix published with `mock-fs@4.10.0` (thanks @3cp).

**Comment by 3cp** [comment]
> `"mock-fs": "git+https://github.com/tschaub/mock-fs.git"`)

FYI, npm/yarn/pnpm all support short-cut of github repo `"mock-fs": "tschaub/mock-fs"` or `"mock-fs": "github:tschaub/mock-fs"`.
